### ISSUE TITLE
Type system confundiendo Strings y Lists

### ISSUE THREAD

**Issue by asanzo** [uqbar-project/wollok#1862]
Me pasa algo raro con este código:

```Wollok
class Revert inherits Commit{
	constructor(cambios, carpeta, descripcion){
		cambios = cambios.reverse().map({cambio => cambio.invertir()})
		descripcion = "REVERT " + descripcion
	}	
}
```

![Selección_999(252)](https://user-images.githubusercontent.com/5421992/67816524-f9812f00-fa88-11e9-9ee3-8a73a8785af0.png)

**Comment by fdodino** [comment]
Perdón por el ruido @asanzo, pero los constructores habíamos definido de no usarlos más (estamos a un paso de sacarlos), los siguen usando a la mañana? Listo, sigo con lo mío...

**Comment by PalumboN** [comment]
No, no damos más constructores. Igual sospecho que este problema pasa más
allá de los constructores.

**Comment by fdodino** [comment]
Sí, mi pregunta era off-topic del issue :smile:

**Comment by isaiaslafon** [comment]
consulta, ¿esto esta relacionado?
[https://drive.google.com/drive/folders/1T6bJADTT3rrQkUwEpT7Sl60j-nCBSv0f?usp=sharing](url)

**Comment by matifreyre** [comment]
Este repo <https://github.com/pdep-sm/2020-clase-objetos-07> tiene 2
"confusiones" similares, la primera es justamente de String y List y la
segunda de Temporada y Capítulo. Calculo que deben estar relacionados, y no
hay constructores en el modelo.

**Comment by fdodino** [comment]
sí, la confusión de String y List<T> viene porque ambos entienden el mensaje `reverse`, pero con el map posterior debería desambiguarse.
